src/extension/libs/stepper/src/stepper.ts

```diff
@@ -0,0 +1,39 @@
+import { IStep } from "./api";
+
+export class Stepper {
+
+    private steps: IStep[] = [];
+
+    public constructor(
+        private title: string
+    ) { }
+
+    public addStep(step: IStep) {
+        this.steps.push(step);
+    }
+
+    public async run<R extends unknown>(): Promise<(R | undefined)[]> {
+
+        const steps: IStep[] = [...this.steps];
+        const result: (R | undefined)[] = [];
+
+        let step = steps.shift();
+
+        while (step) {
+
+            step.title = this.title;
+
+            try {
+                const stepResult = await step.execute();
+                result.push(stepResult as R);
+            } catch (error) {
+                result.push(void 0);
+            }
+
+            step.destroy();
```

Comment:
@Konrad maybe we can think about that to destroy the step on this point, i mean at the end it will remove only events and dispose the field which is good. But maybe we want to get it back should working but then destroy is the wrong name since it is not a real destroy more a dispose.